riscv-pages/src/page.rs

```diff
@@ -393,6 +393,9 @@ pub trait AssignablePhysPage<M: MeasureRequirement>: ConvertedPhysPage {
 /// by unmapping a previously-mapped page in a VM's address space.
 pub trait InvalidatedPhysPage: PhysPage {}
 
+/// Trait representing a mapped page that's eligible for conversion to a Shared state
+pub trait ShareablePhysPage: PhysPage {}
```

Comment:
Add an associated type for the result of the transition to a shared & mappable page?

e.g.

```
pub trait ShareablePhysPage: PhysPage {
    type MappablePage: MappablePhysPage<MeasureOptional>;
}
```

And then, if you take my suggestion on type names, you'd have:

```
impl ShareablePhysPage for Page<Mapped> {
    type MappablePage = Page<MappableShared>;
}
```

Reply:
Done. FWIW, I am finding it extremely difficult to infer the intended purpose of various traits from the code. It's possible that I missed it, but a written documentation explaining the purpose of each of these traits and how are are associated will be very useful.